src/main/java/com/bina/varsim/util/Position1.java

```diff
@@ -0,0 +1,20 @@
+package com.bina.varsim.util;
```

Comment:
that is actually a possible way, but would take an extra integer to store... what's going on here really is that I'm trying to do c++'s template\<int Base\> class Pos{ template\<int OtherBase\> Fcn(const OtherBase& other); }; compile time dispatching. can't figure out how to do it in java.